src/main/java/com/puppycrawl/tools/checkstyle/Checker.java

```diff
@@ -549,13 +549,14 @@ public final void setFileExtensions(String... extensions) {
         }
         else {
             fileExtensions = new String[extensions.length];
+            final String dot = ".";
```

Comment:
Make it class level constant, name it like extension_separator

Reply:
done